promql/parser/prettier.go

```diff
@@ -0,0 +1,557 @@
+			}
+
+			if items[keywordIndex-1].Typ != COMMENT && keywordIndex-1 != aggregatorIndex {
+				formatItems = true
+			} else if items[keywordIndex-1].Typ == COMMENT {
+				// Peek back until no comment is found.
+				// If the item immediately before the comment is not aggregator, that means
+				// the keyword is not (immediately) after aggregator item and hence formatting
+				// is required.
+				var j int
+				for j = keywordIndex; j > aggregatorIndex; j-- {
+					if items[j].Typ == COMMENT {
+						continue
+					}
+					break
+				}
+				if j != aggregatorIndex {
+					formatItems = true
+				}
+			}
+			if !formatItems {
+				continue
+			}
+			for j := keywordIndex; j <= len(items); j++ {
+				if items[j].Typ == RIGHT_PAREN {
+					closingParenIndex = j
+					break
+				}
+			}
+			if closingParenIndex == itemNotFound {
+				return items, errors.New("invalid paren index: closing paren not found")
+			}
+			tempItems := make([]Item, len(items[:aggregatorIndex+1]))
+			// Copy all the items in the items slice from starting till the aggregatorIndex
+			// by value.
+			copy(tempItems, items[:aggregatorIndex+1])
+			// Append items to itemItems from the keyword till the RIGHT_PAREN of the grouping_modifier.
+			tempItems = append(tempItems, items[keywordIndex:closingParenIndex+1]...)
+			// Append the remaining items of the node.
+			for itemp := aggregatorIndex + 1; itemp < len(items); itemp++ {
+				if itemp >= keywordIndex && itemp <= closingParenIndex {
+					continue
+				}
+				tempItems = append(tempItems, items[itemp])
+			}
+			items = tempItems
+
+		case IDENTIFIER, METRIC_IDENTIFIER:
+			// Case-1: {__name__="metric_name"} => metric_name
+			// Case-2: {__name__="metric_name", job="prettier"} => metric_name{job="prettier"}
+			itr := skipComments(items, i)
+			if i < 1 || item.Val != "__name__" || items[itr+2].Typ != STRING {
+				continue
+			}
+			var (
+				leftBraceIndex  = itemNotFound
+				rightBraceIndex = itemNotFound
+				labelValItem    = items[itr+2]
+				metricName      = labelValItem.Val[1 : len(labelValItem.Val)-1] // Trim inverted-commas.
+				tmp             []Item
+				skipBraces      bool // For handling metric_name{} -> metric_name.
+			)
+			if isMetricNameNotAtomic(metricName) {
+				continue
+			}
+			for backScanIndex := itr; backScanIndex >= 0; backScanIndex-- {
+				if items[backScanIndex].Typ == LEFT_BRACE {
+					leftBraceIndex = backScanIndex
+					break
+				}
+			}
+			for forwardScanIndex := itr; forwardScanIndex < len(items); forwardScanIndex++ {
+				if items[forwardScanIndex].Typ == RIGHT_BRACE {
+					rightBraceIndex = forwardScanIndex
+					break
+				}
+			}
+			if leftBraceIndex == itemNotFound || rightBraceIndex == itemNotFound {
+				continue
+			}
+			itr = skipComments(items, itr)
+			if items[itr+3].Typ == COMMA {
+				skipBraces = rightBraceIndex-5 == skipComments(items, leftBraceIndex)
+			} else {
+				skipBraces = rightBraceIndex-4 == skipComments(items, leftBraceIndex)
+			}
```

Comment:
Does it skip braces in case of `{__name__="metric_name",}`?

Reply:
Yes, it does. I have added a unit test for the exact purpose.